woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php

```diff
@@ -98,7 +98,11 @@ public function add_support() {
 		add_filter( 'woocommerce_my_subscriptions_payment_method', array( $this, 'maybe_render_payment_method' ), 10, 3 );
 
 		// don't copy over order-specific meta to the WC_Subscription object during renewal processing
-		add_filter( 'wcs_renewal_order_meta', array( $this, 'do_not_copy_order_meta' ) );
+		if ( SV_WC_Plugin_Compatibility::is_wc_subscriptions_version_gte( '2.5' ) ) {
+			add_filter( 'wc_subscriptions_renewal_order_data', [ $this, 'do_not_copy_order_meta' ] );
+		} else {
+			add_filter( 'wcs_renewal_order_meta', [ $this, 'do_not_copy_order_meta' ] );
```

Comment:
Thanks for making this move. I also tried just renaming callback, but this triggered a bunch of undefined meta key PHP warnings. API for these callbacks maybe different, is this covered by tests?

Reply:
hey @lkraav thanks for checking this out!

I was under the impression the filter had the same signature as declared in the line I pointed it out... 

May I ask which plugins are you using that rely on this FW filter?

Reply:
maybe the issue is in the way the $order_meta array is shaped https://github.com/godaddy-wordpress/wc-plugin-framework/blob/1f9d73cb87732aa35172125c0e3130a22d29a871/woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php#L448 and that meta_key no longer exists?

Reply:
I think  https://github.com/godaddy-wordpress/wc-plugin-framework/pull/642/commits/b37337593b1c5e57eaa6ba731e8352f506c66e22 might do the trick -- according to the updated filter description:

>  The data to be copied to the "to" object. Each value is keyed by the meta key. Example format [ '_meta_key' => 'meta_value' ]

needs testing with some concrete object, but the commit handles each possible array shape

Reply:
> May I ask which plugins are you using that rely on this FW filter?

PayPal Gateway Powered by Braintree

https://plugins.trac.wordpress.org/browser/woocommerce-gateway-paypal-powered-by-braintree/tags/3.0.8/vendor/skyverge/wc-plugin-framework/woocommerce

Reply:
ah I see, that's maintained by A8C now -- the WC team will have to update the FW in that one after we push this change